cpptod.dd

```diff
@@ -913,6 +913,43 @@ SecondVariant.method();
 ------
 
 
+<hr>$(COMMENT  -------------------------------------------- )
+$(H3 <a name="structcmp">References in D</a>)
+
+        D doesn't have a (C++-style) concept of references as part of the type. Arguments can be passed by reference - hence the `ref` keyword, but "free" references don't exist in the language.
+
+        The ref in foreach loop variables can be conceptually thought of as a parameter to the loop body as well. (For opApply-based iteration, the loop body indeed gets turned into a function; for "plain" iteration the compiler AST internally has special ref variables, but they are not visible to the language.)
+
+        In the below code, d2 is a value copy of gallery[0] and is not a reference.
+
```

Comment:
`$(RUNNABLE_EXAMPLE` and close the parentheses after the ending `---`
This will allow people to execute this code - like on the documentation.

Reply:
Thanks